src/haz3lcore/dynamics/Builtins.re

```diff
@@ -33,7 +33,27 @@ module Pervasives = {
   module Impls = {
     open EvaluatorMonad;
     open EvaluatorResult;
-
+    /* int_of_string implementation. */
+    let int_of_string = (name, r1) =>
+      switch (r1) {
+      | BoxedValue(StringLit(f) as d1) =>
+        let f = Re.Str.string_after(f, 1);
+        let f = Re.Str.string_before(f, String.length(f) - 1);
+        if (String.length(f) == 0) {
+          BoxedValue(IntLit(0)) |> return;
+        } else if (Form.is_int(f)) {
+          let i = int_of_string(f);
+          BoxedValue(IntLit(i)) |> return;
+        } else {
+          Indet(
+            InvalidOperation(ApBuiltin(name, [d1]), InvalidIntOfString),
```

Comment:
here I use the invalid operator to indicate the invalid string to int conversion.